`hepmc3`: fix typo in cmake arg for the `+protobuf` variant

luketpickering opened the pull request:
c.f. https://gitlab.cern.ch/hepmc/HepMC3/-/blob/master/CMakeLists.txt?ref_type=heads#L32

luketpickering committed:
fix typo in variable name in hepmc3 variant

luketpickering committed:
set cxx standard to 14 when using protobuf

luketpickering committed:
add myself to hepmc3 maintainer list

alecbcs commented on var/spack/repos/builtin/packages/hepmc3/package.py:
```diff
@@ -85,4 +85,7 @@ def cmake_args(self):
                     self.define("HEPMC3_CXX_STANDARD", spec["root"].variants["cxxstd"].value)
                 )
 
+        if "+protobuf" in spec:
+            args.append(self.define("HEPMC3_CXX_STANDARD", "14"))
```
```suggestion
        if spec.satisfies("+protobuf"):
            args.append(self.define("HEPMC3_CXX_STANDARD", "14"))
```

It is now recommended to use `satisfies()` instead of `in` for checking spec attributes.

bernhardkaindl committed:
hepmc3: Applied suggestion of @acecbs for spec.satisfies("+protobuf") (agreed!)

Co-authored-by: Alec Scott <hi@alecbcs.com>

sethrj commented on var/spack/repos/builtin/packages/hepmc3/package.py:
```diff
@@ -85,4 +85,7 @@ def cmake_args(self):
                 )
 
+        if spec.satisfies("+protobuf"):
+            args.append(self.define("HEPMC3_CXX_STANDARD", "14"))
```
Perhaps this should be an `elif`, otherwise it will collide with the ROOT definition? Perhaps also we should add a `cxxstd` variant that must match ROOT (if there) or be at least 14 if protobuf is included.

luketpickering replied:
yep, good point, agreed. I suspect any recent build including ROOT will set the standard to 17 at least, which obviates this. Its only for people (me) who want protobuf and not root support that it shows up. The real fix is on the HepMC3 cmake side, which I'll try and do when I have a spare cycle, but for now I think the elif covers most cases and still fixes my specific error.

Thanks!

sethrj reviewed (commented):
Good catch with the CMake argument name! I think we just need to figure out how to set `cxxstd` correctly.

luketpickering committed:
Merge branch 'spack:develop' into patch-1

luketpickering committed:
hepmc3: cxx_standard for protobuf

only set cxx standard to meet protobuf minimum (14) if not also rootio variant as that sets the cxx standard to match the root public API standard requirements
